### ISSUE TITLE
Support changing the endpoint of Octo STS for Self hosting

### ISSUE THREAD

**Issue by suzuki-shunsuke** [octo-sts/action#26]
Now the endpoint of Octo STS is hardcoded to `https://octo-sts.dev/sts/exchange`.

https://github.com/octo-sts/action/blob/3622e4d541c8adbc31c21017a7025ef418b94c09/index.js#L42

To use this action with self-hosted Octo STS, we need to change the endpoint.
So how about adding an optional input variable `endpoint` to change the endpoint?
The default value is `https://octo-sts.dev/sts/exchange`.

e.g.

```yaml
- uses: octo-sts/action@main
  id: octo-sts
  with:
    scope: your-org/your-repo
    identity: foo
    endpoint: https://example.com/sts/exchange
```

**Comment by suzuki-shunsuke** [comment]
I'll send a pull request soon.

**Comment by tuananh** [comment]
I think it maybe not enough as i saw `octo-sts.dev` hardcoded in octo-sts in quite many places.

**Comment by suzuki-shunsuke** [comment]
📝

> I think it maybe not enough as i saw `octo-sts.dev` hardcoded in octo-sts in quite many places.

https://github.com/search?q=repo%3Aocto-sts%2Fapp%20octo-sts.dev&type=code

```console
~/repos/src/github.com/octo-sts/app main
$ git grep octo-sts.dev 
README.md:  "https://octo-sts.dev/sts/exchange?scope=${REPO}&identity=${NAME}"
iac/gclb.tf:  name        = "octo-sts-dev"
iac/gclb.tf:  dns_name    = "octo-sts.dev."
iac/gclb.tf:  description = "DNS zone for domain: octo-sts.dev"
iac/gclb.tf:    "octo-sts.dev" = {
pkg/octosts/octosts.go:         event.SetSource("https://octo-sts.dev")
pkg/octosts/octosts.go:         ClientID: "octo-sts.dev",
pkg/prober/prober.go:           "https://octo-sts.dev",
pkg/prober/prober.go:   ts, err := idtoken.NewTokenSource(ctx, "octo-sts.dev" /* aud */)
pkg/prober/prober.go:           "https://octo-sts.dev",
pkg/prober/prober.go:           "https://octo-sts.dev",
pkg/prober/prober.go:   ts, err := idtoken.NewTokenSource(ctx, "octo-sts.dev" /* aud */)
```

**Comment by mkushakov** [comment]
Looks like issue was already fixed by #28

**Comment by suzuki-shunsuke** [comment]
> Looks like issue was already fixed by #28
